network-config-analyzer/IstioNetworkPolicy.py

```diff
@@ -0,0 +1,188 @@
+        self.action = IstioNetworkPolicy.ActionType.Allow
+
+    def __eq__(self, other):
+        return super().__eq__(other) and self.action == other.action
+
+    # 'deny' policies should be evaluated before 'allow' policies
+    def __lt__(self, other):  # required so we can evaluate the policies according to their order
+        if isinstance(other, IstioNetworkPolicy):
+            return self.action == IstioNetworkPolicy.ActionType.Deny
+        return NotImplemented
+
+    def allowed_connections(self, from_peer, to_peer, is_ingress):
+        """
+        Evaluate the set of connections this policy allows/denies/passes between two peers
+        :param Peer.Peer from_peer: The source peer
+        :param Peer.Peer to_peer:  The target peer
+        :param bool is_ingress: whether we evaluate ingress rules only or egress rules only
+        :return: A PolicyConnections object containing sets of allowed/denied/pass connections
+        :rtype: PolicyConnections
+        """
+
+        # TODO: currently not handling egress, istio authorization policies have no egress rules
+        if not is_ingress:
+            return PolicyConnections(False, ConnectionSet(True))
+
+        captured = to_peer in self.selected_peers
+        if not captured:
+            return PolicyConnections(False)
+
+        allowed_conns = ConnectionSet()
+        denied_conns = ConnectionSet()
+
+        rules = self.ingress_rules
+        collected_conns = allowed_conns if self.action == IstioNetworkPolicy.ActionType.Allow else denied_conns
+        for rule in rules:
+            if from_peer in rule.peer_set:
+                rule_conns = rule.connections
+                collected_conns |= rule_conns
+
+        return PolicyConnections(True, allowed_conns, denied_conns)
+
+    def referenced_ip_blocks(self):
+        """
+        :return: A set of all ipblocks referenced in one of the policy rules (one Peer object per one ip range)
+        :rtype: Peer.PeerSet
+        """
+        res = PeerSet()
+        for rule in self.ingress_rules:
+            for peer in rule.peer_set:
+                if isinstance(peer, IpBlock):
+                    res |= peer.split()
+        return res
+
+    def has_empty_rules(self, config_name=''):
+        """
+        Checks whether the policy contains empty rules (rules that do not select any peers)
+        :param str config_name: (optional) the name of the NetworkConfig object
+        :return: A list of strings describing the emptiness + two sets of indexes of empty ingress/egress rules
+        :rtype: list[str], set, set
+        """
+        emptiness_explanation = []
+        empty_ingress_rules = set()
+        full_name = self.full_name(config_name)
+        for rule_index, ingress_rule in enumerate(self.ingress_rules, start=1):
+            if not ingress_rule.peer_set:
+                emptiness = f'Ingress rule no. {rule_index} in AuthorizationPolicy {full_name} does not select any pods'
```

Comment:
Should we leave the `Ingress` here? It is kind of obvious

Reply:
done